server/index.js

```diff
@@ -49,6 +50,31 @@ app.post('/api/bookShelf', (req, res, next) => {
     })
     .catch(err => next(err));
 });
+
+app.patch('/api/bookShelf/library/:googleId', (req, res, next) => {
+  const googleId = req.params.googleId;
+  if (!googleId) {
+    res.status(400).json({
+      error: 'googleId is required'
+    });
+    return;
+  }
```

Comment:
This route would not have been matched if there were fewer than four path segments, so you can remove this `if`.